File: backend/app/api/study.py
from fastapi import APIRouter

from app.schemas.study import StudyGenerateRequest, StudyGenerateResponse
from app.services.study.study_generator import generate_study_activity

router = APIRouter()


@router.post("/generate", response_model=StudyGenerateResponse)
def generate_study(request: StudyGenerateRequest) -> StudyGenerateResponse:
    result = generate_study_activity(
        collection_id=request.collection_id,
        document_id=request.document_id,
        mode=request.mode,
        count=request.count,
        quality=request.quality,
    )
    return StudyGenerateResponse(**result)

File: frontend/src/api/study.ts
import { StudyGenerateResponse, StudyMode } from "./types";
import { requestJson } from "./client";

export async function generateStudy(
  collectionId: string,
  mode: StudyMode,
  quality: boolean,
  count?: number
): Promise<StudyGenerateResponse> {
  return requestJson<StudyGenerateResponse>("/api/study/generate", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ collection_id: collectionId, mode, count, quality })
  });
}
